### PR TITLE
restore the purple line

### PR DESCRIPTION
**Pull request by jasoncalabrese**
forecasts

### CODE REVIEW

File: lib/plugins/openaps.js [hunk 1]
```diff
@@ -286,6 +293,22 @@ function init() {
...
+      var points = null;
+      if (prop.lastPredBGs) {
+        points = _.map(prop.lastPredBGs.values, function toPoint(value, index) {
+          return {
+            mgdl: value
+            , color: '#ff00ff'
+            , mills: prop.lastPredBGs.moment.valueOf() + times.mins(5 * index).msecs
+            , noFade: true
+          }
+        })
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2403546780/issues/source?bid=2452379&fileBranchId=3233006#l306)

File: lib/plugins/openaps.js [hunk 2]
```diff
@@ -286,6 +293,22 @@ function init() {
       return valueParts;
     }
 
+    function getForecastPoints ( ) {
+      var points = null;
+      if (prop.lastPredBGs) {
+        points = _.map(prop.lastPredBGs.values, function toPoint(value, index) {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2403546780/issues/source?bid=2452379&fileBranchId=3233006#l299)

File: lib/plugins/openaps.js [hunk 3]
```diff
@@ -286,6 +293,22 @@ function init() {
...
+          return {
+            mgdl: value
+            , color: '#ff00ff'
+            , mills: prop.lastPredBGs.moment.valueOf() + times.mins(5 * index).msecs
+            , noFade: true
+          }
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2403546780/issues/source?bid=2452379&fileBranchId=3233006#l305)

File: lib/plugins/openaps.js [hunk 4]
```diff
@@ -286,6 +293,22 @@ function init() {
...
+      var points = null;
+      if (prop.lastPredBGs) {
+        points = _.map(prop.lastPredBGs.values, function toPoint (value, index) {
+          return {
+            mgdl: value
+            , color: '#ff00ff'
+            , mills: prop.lastPredBGs.moment.valueOf() + times.mins(5 * index).msecs
+            , noFade: true
+          };
+        })
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2403555791/issues/source?bid=2452379&fileBranchId=3233006#l306)

File: lib/plugins/openaps.js [hunk 5]
```diff
@@ -286,6 +295,36 @@ function init() {
...
+
+      if (prop.lastPredBGs) {
+        if (prop.lastPredBGs.values) {
+          points = points.concat(_.map(prop.lastPredBGs.values, toPoint))
+        }
+        if (prop.lastPredBGs.IOB) {
+          points = points.concat(_.map(prop.lastPredBGs.IOB, toPoint))
+        }
+        if (prop.lastPredBGs.aCOB) {
+          points = points.concat(_.map(prop.lastPredBGs.aCOB, toPoint))
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2404431938/issues/source?bid=2452379&fileBranchId=3233006#l318)

File: lib/plugins/openaps.js [hunk 6]
```diff
@@ -286,6 +295,36 @@ function init() {
...
+          , noFade: true
+        };
+      }
+
+      if (prop.lastPredBGs) {
+        if (prop.lastPredBGs.values) {
+          points = points.concat(_.map(prop.lastPredBGs.values, toPoint))
+        }
+        if (prop.lastPredBGs.IOB) {
+          points = points.concat(_.map(prop.lastPredBGs.IOB, toPoint))
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2404431938/issues/source?bid=2452379&fileBranchId=3233006#l315)

File: lib/plugins/openaps.js [hunk 7]
```diff
@@ -286,6 +295,36 @@ function init() {
...
+          mgdl: value
+          , color: '#ff00ff'
+          , mills: prop.lastPredBGs.moment.valueOf() + times.mins(5 * index).msecs
+          , noFade: true
+        };
+      }
+
+      if (prop.lastPredBGs) {
+        if (prop.lastPredBGs.values) {
+          points = points.concat(_.map(prop.lastPredBGs.values, toPoint))
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2404431938/issues/source?bid=2452379&fileBranchId=3233006#l312)

File: tests/openaps.test.js [hunk 8]
```diff
@@ -141,6 +146,9 @@ describe('openaps', function ( ) {
           var last = _.last(options.info);
           last.label.should.equal('1h ago');
           last.value.should.equal('awaitingpi ◉ Waiting');
+        }
+        , addForecastPoints: function mockAddForecastPoints(points) {
```
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2416383334/issues/source?bid=2452379&fileBranchId=3233006#l150)

File: tests/openaps.test.js [hunk 9]
```diff
@@ -126,7 +131,7 @@ _.forEach(statuses, function updateMills (status) {
 
 describe('openaps', function ( ) {
 
-  it('set the property and update the pill', function (done) {
+  it('set the property and update the pill and add forecast points', function (done) {
```
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2416383334/issues/source?bid=2452379&fileBranchId=3233006#l134)

**Comment by jasoncalabrese** [comment]
@MilosKozak I'm thinking about using this view menu for more, might be a good place to add the hide context that you've wanted.  Maybe edit mode too?

**Comment by MilosKozak** [comment]
right click on some specific place?

**Comment by MilosKozak** [comment]
i'd like it much for switching IOB source

File: lib/client/careportal.js [hunk 10]
```diff
@@ -82,16 +93,45 @@ function init (client, plugins, $) {
...
+
     maybePrevent(event);
   };
 
+  careportal.reasonable = function reasonable ( ) {
+    var eventType = $('#eventType').val();
+    var reasons = inputMatrix[eventType]['reasons'];
+    var selected = $('#reason').val();
+
+    var reason = _.find(reasons, function matches(r) {
```
Comment (jasoncalabrese) [inline lib/client/careportal.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417624266/issues/source?bid=2452379&fileBranchId=3233006#l105)

File: lib/client/careportal.js [hunk 11]
```diff
@@ -73,7 +77,14 @@ function init (client, plugins, $) {
     $('#profileLabel').css('display',displayType(inputMatrix[eventType]['profile']));
     $('#preBolusLabel').css('display',displayType(inputMatrix[eventType]['prebolus']));
     $('#insulinSplitLabel').css('display',displayType(inputMatrix[eventType]['split']));
- 
+
+    $('#reason').empty();
+    _.each(reasons, function eachReason(reason) {
```
Comment (jasoncalabrese) [inline lib/client/careportal.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417624266/issues/source?bid=2452379&fileBranchId=3233006#l82)

File: tests/openaps.test.js (hunk 8 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417624382/issues/source?bid=2452379&fileBranchId=3233006#l150)

File: tests/openaps.test.js (hunk 9 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417624382/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/careportal.js (hunk 10 above)
Comment (jasoncalabrese) [inline lib/client/careportal.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417663829/issues/source?bid=2452379&fileBranchId=3233006#l105)

File: lib/client/careportal.js (hunk 11 above)
Comment (jasoncalabrese) [inline lib/client/careportal.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417663829/issues/source?bid=2452379&fileBranchId=3233006#l82)

File: tests/openaps.test.js (hunk 8 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417663945/issues/source?bid=2452379&fileBranchId=3233006#l150)

File: tests/openaps.test.js (hunk 9 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417663945/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/renderer.js [hunk 12]
```diff
@@ -168,8 +173,11 @@ function init (client, d3) {
     function treatmentTooltip (d) {
```
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417663834/issues/source?bid=2452379&fileBranchId=3233006#l173)

File: tests/openaps.test.js [hunk 13]
```diff
@@ -126,13 +131,13 @@ _.forEach(statuses, function updateMills (status) {
...
```
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417700365/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/renderer.js (hunk 12 above)
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2417700254/issues/source?bid=2452379&fileBranchId=3233006#l173)

File: tests/openaps.test.js (hunk 13 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418560118/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/renderer.js (hunk 12 above)
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418560007/issues/source?bid=2452379&fileBranchId=3233006#l173)

File: tests/openaps.test.js (hunk 13 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418618251/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/renderer.js (hunk 12 above)
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418618140/issues/source?bid=2452379&fileBranchId=3233006#l173)

File: tests/openaps.test.js (hunk 13 above)
Comment (jasoncalabrese) [inline tests/openaps.test.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418618503/issues/source?bid=2452379&fileBranchId=3233006#l134)

File: lib/client/renderer.js (hunk 12 above)
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2418618392/issues/source?bid=2452379&fileBranchId=3233006#l173)

File: .travis.yml [hunk 14]
```diff
@@ -1,10 +1,15 @@
...
-  - "0.12"
-  - "4"
-  - "5"
+  - 0.10
+  - 0.12
+  - 4
+  - 5
+matrix:
+  fast_finish: true
+  allow_failures:
```
Comment (jasoncalabrese) [inline .travis.yml]: @MilosKozak this should help with some of the build failures

**Pull request merged**